compiler/src/iree/compiler/Dialect/Stream/IR/test/async_folding.mlir

```diff
@@ -54,23 +54,25 @@ func.func @SplatAlreadyAtSinkLocation(
   %c2 = arith.constant 2 : index
   %c3 = arith.constant 3 : index
   %c100 = arith.constant 100 : index
+  %c101 = arith.constant 101 : index
   %c121_i32 = arith.constant 121 : i32
   // The splat is already where we would sink it to -- this used to trigger
-  // a bug where we would "move" the splat to its current location, triggering
   // infinite pattern recursion.
   // CHECK: %[[SPLAT:.+]] = stream.async.splat %c121_i32 : i32 -> !stream.resource<*>{%c100}
+  // CHECK-NEXT: %[[SPLAT:.+]] = stream.async.splat %c121_i32 : i32 -> !stream.resource<*>{%c101}
   // CHECK-NEXT: cf.cond_br %arg1, ^bb1, ^bb2
   %0 = stream.async.splat %c121_i32 : i32 -> !stream.resource<*>{%c100}
+  %1 = stream.async.splat %c121_i32 : i32 -> !stream.resource<*>{%c101}
   cf.cond_br %arg1, ^bb1, ^bb2
 // CHECK: ^bb1:
 ^bb1:
-  // CHECK: = stream.async.dispatch @executable::@dispatch0[%c1, %c2, %c3](%[[SPLAT]][%c0 to %c100 for %c100])
-  %2 = stream.async.dispatch @executable::@dispatch0[%c1, %c2, %c3](%0[%c0 to %c100 for %c100]) : (!stream.resource<*>{%c100}) -> !stream.resource<*>{%c100}
+  // CHECK: stream.async.dispatch @executable::@dispatch0[%c1, %c2, %c3](%0[%c0 to %c100 for %c100], %1[%c0 to %c101 for %c101]) : (!stream.resource<*>{%c100}, !stream.resource<*>{%c101}) -> !stream.resource<*>{%c100}
+  %2 = stream.async.dispatch @executable::@dispatch0[%c1, %c2, %c3](%0[%c0 to %c100 for %c100], %1[%c0 to %c101 for %c101]) : (!stream.resource<*>{%c100}, !stream.resource<*>{%c101}) -> !stream.resource<*>{%c100}
   cf.br ^bb3(%2 : !stream.resource<*>)
 // CHECK: ^bb2:
 ^bb2:
-  // CHECK: = stream.async.dispatch @executable::@dispatch1[%c1, %c2, %c3](%[[SPLAT]][%c0 to %c100 for %c100])
-  %3 = stream.async.dispatch @executable::@dispatch1[%c1, %c2, %c3](%0[%c0 to %c100 for %c100]) : (!stream.resource<*>{%c100}) -> !stream.resource<*>{%c100}
+  // CHECK: stream.async.dispatch @executable::@dispatch1[%c1, %c2, %c3](%0[%c0 to %c100 for %c100], %1[%c0 to %c101 for %c101]) : (!stream.resource<*>{%c100}, !stream.resource<*>{%c101}) -> !stream.resource<*>{%c100}
+  %3 = stream.async.dispatch @executable::@dispatch1[%c1, %c2, %c3](%0[%c0 to %c100 for %c100], %1[%c0 to %c101 for %c101]) : (!stream.resource<*>{%c100}, !stream.resource<*>{%c101}) -> !stream.resource<*>{%c100}
```

Comment:
can you please switch to using named captures instead of %0 or %1? (SPLAT100/SPLAT101, etc)
(both splats are captured as SPLAT here)

Reply:
Good catch!